### PR TITLE
 Updated ast.MemberExpr, equal, and added more tests

### PR DESCRIPTION
**Pull request by MichaelS11**

### CODE REVIEW

**Comment by MichaelS11** [comment]
@mattn ,

Any thoughts on this?

**Comment by mattn** [comment]
Please note what you want to change. AFAICS, you seems to want to change `nil == nil` and add `kindOf` ?

**Comment by MichaelS11** [comment]
Added toInt, updated error codes, changed so return can be used in main, added kindOf, updated member expressions and index expressions. Those are the bigger changes. The update to IsValid seemed fairly minor, not sure if it really changed much.

File: builtins/regexp/regexp.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 func Import(env *vm.Env) *vm.Env {
-	m := env.NewPackage("sort")
+	m := env.NewPackage("regexp")
```
Comment (mattn) [inline builtins/regexp/regexp.go]: good catch. thank you

Comment (MichaelS11) [reply builtins/regexp/regexp.go]: Welcome :)

**Comment by MichaelS11** [comment]
Note that the changes are grouped into there own commit. The addition of new tests or test changes are also in there own commit. The hope was that would make it easier to review. Please let me know if there is anything else I can do to help make the changes easier to review.

**Comment by mattn** [comment]
LGTM. Thank you. In next time for PR, please note what you want to change in description.

**Pull request merged**